packages/workflow/src/worker-interface.ts

```diff
@@ -189,8 +189,18 @@ export function activate(activation: coresdk.workflow_activation.WorkflowActivat
       if (activation.historyLength == null) {
         throw new TypeError('Got activation with no historyLength');
       }
+      if (activation.historySizeBytes == null) {
```

Comment:
@Sushisource What will we get at this point in `activation.historySizeBytes` and `activation.continueAsNewSuggested` if we process a WFT dating back from before these fields were added into server? Is it safe to assume the fact core will have filled these props with non-null values?

Reply:
I was tempted to remove the check, in sdk-core/core/src/worker/workflow/machines/workflow_machines.rs:250 is set to 0, and then it uses `unwrap_or_default()`, so it should not be null
 @Sushisource can you confirm this?

Reply:
Well since these come across to lang as protos, they're just gonna be zeroed out / false if there's no real value to read. Whether or not they are null depends on how the TS bridge converts Rust structures to TS. Nulls don't exist in Rust, of course, and these fields are not `Option`s, so I can't imagine why they'd ever get converted to null

Reply:
Right... 🤔 Seems then like a few of the checks around here are totally useless. `historyLength`, `historySizeBytes` and `continueAsNewSuggested` can't possibly be null or undefined, as they are all primitives from in the protobuf definition, so they would anyway return their default value if missing. Same for `isReplaying` a few lines below.

Reply:
Makes sense. The same for continueAsNewSuggested. As long as these values are going through core there will always have a default value. The only exception is unit tests that handcraft the activations... @mjameswh are you OK for me to remove them?

Reply:
> As long as these values are going through core there will always have a default value. 

To be exact, both Core _and_ Protobufjs are providing us guarantees here.

> are you OK for me to remove them?

Yeah. If we're sure they don't bring any actual value, then I'd prefer not having such checks. They led us in the wrong direction this time, and could very likely lead us in a wrong direction again in the future if we don't get rid of them.

For unit tests, we could simply reencode handcrafted activations objects into actual protobufjs objects. Something like this this (not sure what is the difference between `create` and `fromObject`):

```
async function activate(t: ExecutionContext<Context>, activation: coresdk.workflow_activation.IWorkflowActivation) {
  const act = coresdk.workflow_activation.WorkflowActivation.fromObject(activation);
  // ...
```

Reply:
One catch with protobufjs is that they add the default values to the prototype and then Object#hasOwnProperty (and for of) does not work protobufjs/protobuf.js/#521 .  Is this going to be an issue for client/visibility API?

Reply:
I don't see any potential issue here. Activations are only meant for internal usage, and that change would only be for unit tests anyway.

Reply:
Checks removed